snuba/query/parser/strings.py

```diff
@@ -0,0 +1,36 @@
+from snuba.datasets.tags_column_processor import NESTED_COL_EXPR_RE
+from snuba.query.expressions import Column, Expression, Literal, SubscriptableReference
+from snuba.util import QUOTED_LITERAL_RE
+
+
+def parse_string_to_expr(val: str) -> Expression:
+    """
+    Parses a string expression found in a Snuba Query into an AST expression.
+    This is supposed to work for any non structured expression in the Snuba Query.
+    """
+    match = NESTED_COL_EXPR_RE.match(val)
+    if match:
+        col_name = match[1]
+        key_name = match[2]
+        return SubscriptableReference(
+            # TODO: apply aliases on all AST nodes created during parsing and that do not
+            # already have one. This makes the field names from the original query
+            # (that the client expects in the response) available when building the
+            # Clickhouse query without leaking the logical schema.
```

Comment:
I'm not sure that I understand this comment.

Reply:
Yeah, this was quite hard to phrase. Anyway this is an example:
- the user provides a query that includes `tags[asd]` in the query without an alias.
- the response needs to include (as it happens today) `whatever_expression_tag_gets_resolved_to() AS tags[asd]`  So that the user can reference that specific column.
- this implies we need to propagate across the query processing pipeline the original field name the user provided in the query `tags[asd]` so it can be added as alias in the clickhouse query
- I think the clean way of doing so is to make the parser apply aliases to all expressions referenced in the select clause that don;t already have them. The consequence is that, no matter what we do to those expressions during query processing those aliases will be passed to clickhouse and thus present in the response.

Reply:
Ok — that makes sense, thanks for clarifying.

I was thinking about this earlier as part of query translation but reading this it makes more sense to deal with this as soon as possible so other logical processors preserve aliases as much as possible.